src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityLongDistanceEndpoint.java

```diff
@@ -0,0 +1,226 @@
+package gregtech.common.metatileentities.storage;
+
+import gregtech.api.metatileentity.IDataInfoProvider;
+import gregtech.api.metatileentity.MetaTileEntity;
+import gregtech.api.pipenet.longdist.ILDEndpoint;
+import gregtech.api.pipenet.longdist.LongDistanceNetwork;
+import gregtech.api.pipenet.longdist.LongDistancePipeType;
+import gregtech.common.ConfigHolder;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraft.world.World;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class MetaTileEntityLongDistanceEndpoint extends MetaTileEntity implements ILDEndpoint, IDataInfoProvider {
+
+    private final LongDistancePipeType pipeType;
+    private Type type = Type.NONE;
+    private ILDEndpoint link;
+    private boolean placed = false;
+
+    public MetaTileEntityLongDistanceEndpoint(ResourceLocation metaTileEntityId, LongDistancePipeType pipeType) {
+        super(metaTileEntityId);
+        this.pipeType = pipeType;
+    }
+
+    public void updateNetwork() {
+        LongDistanceNetwork network = LongDistanceNetwork.get(getWorld(), getPos());
+        if (network != null) {
+            int endpointAmount = network.getEndpointAmount();
+            if (endpointAmount == 2) return;
+
+            if (endpointAmount == 1) {
+                network.onPlaceEndpoint(this);
+                return;
+            }
+            network.onRemoveEndpoint(this);
```

Comment:
these two if statements return too early and could be simplified with 
```
if (endpointAmount == 1) {
    network.onPlaceEndpoint(this);
} else if (endpointAmount == 2) {
    network.onRemoveEndpoint(this);
}
```

Reply:
I can not reproduce that issue. Do you have a setup?

Reply:
i just had a straight LD item pipe about 53 blocks long, then tried reversing the inputs and outputs.

Reply:
this issue was fixed by 11f5fdc